lib/plugins/features/movies/controller.js

```diff
@@ -34,3 +35,16 @@ exports.list = async (query) => {
 
   }).fetchAll({ withRelated: ['locations'] });
 };
+
+exports.addLocationToMovie = async (movie_id, location_id) => {
+  const movie = await new Movie({ id: movie_id }).fetch({ withRelated: ['locations'] });
+  const location = await new Location({ id: location_id }).fetch();
+
+  const existingLocations = movie.related('locations').models.map((loc) => loc.id);
```

Comment:
based upon my earlier comment, it might be possible to make this more streamlined: we could pass in the `movie_id` and payload containing the location information. We then check to see if a movie exists. If it doesn't, surface an error to the user. If it does, we want to then check to see if the location exists. If it doesn't, create it. If it does or once it's created, go ahead and attach it to the movie. Finally, return the movie with the associated locations.